### PR TITLE
Implement detection and print of scopes for `debug`

### PR DESCRIPTION
**Pull request by AlexandrHoroshih**
# Changelog

- Added detection of forked scopes and handler to register scope to `debug`

### CODE REVIEW

**Review by igorkamyshev — APPROVED** [review]
Офигенно!

**Review by sergeysova — APPROVED** [review]

File: src/debug/scope-cache.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+const cache = new Map<Scope, Meta>();
+
+let unknownScopes = 0;
+function getDefaultName() {
+  unknownScopes += 1;
+  return `unknown_scope_${unknownScopes}`;
+}
+const scopes = {
```
Comment (sergeysova) [inline src/debug/scope-cache.ts:14]:
```suggestion
export const scopes = {
```

File: src/debug/scope-cache.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  },
+  forEach(callback: (scope: Scope, meta: Meta) => void) {
+    cache.forEach((meta, scope) => callback(scope, meta));
+  },
+  clear() {
+    cache.clear();
+  },
+} as const;
+
+export { scopes };
```
Comment (sergeysova) [inline src/debug/scope-cache.ts:35]:
```suggestion
```

File: src/debug/scope-cache.ts (hunk 1 above)
Comment (sergeysova) [reply src/debug/scope-cache.ts:14]: Why not move it inside `debug/index.ts` ?

**Pull request merged**